src/crypto/pubkeyEncryptor.ts

```diff
@@ -0,0 +1,43 @@
+import nacl from "tweetnacl";
+import ed2curve from "ed2curve";
+import crypto from "crypto";
+import { X25519EncryptedData } from "./x25519EncryptedData";
+import { UserPublicKey, UserSecretKey } from "../userKeys";
+import { PubKeyEncCipher, PubKeyEncNonceLength, PubKeyEncVersion } from "./constants";
+
+export class PubkeyEncryptor {
+    static encrypt(data: Buffer, recipientPubKey: UserPublicKey, authSecretKey: UserSecretKey): X25519EncryptedData {
+        // create a new x225519 keypair that will be used for EDH
+        const edhPair = nacl.sign.keyPair();
+        const recipientDHPubKey = ed2curve.convertPublicKey(recipientPubKey.valueOf());
+        if (recipientDHPubKey === null) {
+            throw new Error("Could not convert ed25519 public key to x25519");
+        }
+        const edhConvertedSecretKey = ed2curve.convertSecretKey(edhPair.secretKey);
+
+        const nonce = nacl.randomBytes(PubKeyEncNonceLength);
```

Comment:
should make the nonce static for identical messages?
(the limitation is that the nonce needs to be unique for distinct messages).

Reply:
done

Reply:
@AdoAdoAdo @ccorcoveanu, perhaps use random data, though? So that two identical confidential messages encrypted by the same user would result into different representations (for better security etc.).